### DISCUSSION TITLE
Performance difference according to version

### DISCUSSION THREAD

**Discussion by jiwoonjeon-mcns** [ultralytics/yolov5#9440] [Q&A] [answered]
The results of learning with the previous version (about two months ago) and the results of learning with the current version are very different.(Trained under the same condition) Anyone have the same problem? (Previous version works better)
![results](https://user-images.githubusercontent.com/74587346/190610638-b1fefe02-58d7-4ef6-b7ab-8b00f6c34070.png)

**Comment by glenn-jocher** [comment, marked as answer]
@jiwoonjeon-mcns it's likely that you trained a model differently between the two times (user error). 

If you can create a reproducible example showing different performance on a common dataset (i.e. VOC, COCO128, etc.) with different commits then I'll take a look.

**Reply by jiwoonjeon-mcns** [reply]
I used the yolov5l model and the config, hyp, and dataset are the same. I don't know the cause, so I'm only using the old version.

**Reply by glenn-jocher** [reply]
@jiwoonjeon-mcns that's exactly what I mean, if you can provide **exact code to reproduce on a common dataset** your issue then we can look at it. If not there is no action for us to take. Exact code would be:

```
git clone https://github.com/ultralytics/yolov5 yolov5-before
cd yolov5-before
git checkout HASH
python train.py --data VOC.yaml ARGS

cd ..

git clone https://github.com/ultralytics/yolov5 yolov5-master
cd yolov5-master
python train.py --data VOC.yaml ARGS
```

**Reply by jiwoonjeon-mcns** [reply]
Ok. I will test it with an open dataset.

**Reply by jiwoonjeon-mcns** [reply]
![voc_result](https://user-images.githubusercontent.com/74587346/191391947-0ea8d6ec-ac46-47f2-a27c-c1a989bace1b.png)

This is results for voc dataset. Purple line is the old version.

command
python train.py --data VOC.yaml

**Reply by glenn-jocher** [reply]
@jiwoonjeon-mcns what's the commit hash for older version? I need this code to reproduce like I asked for before:

```bash
git clone https://github.com/ultralytics/yolov5 yolov5-before
cd yolov5-before
git checkout f17444abcd647a299f23fe2bf6324b8947cdee22
python train.py --data VOC.yaml --epochs 50 --cache

cd ..

git clone https://github.com/ultralytics/yolov5 yolov5-master
cd yolov5-master
python train.py --data VOC.yaml  --epochs 50 --cache
```

**Reply by jiwoonjeon-mcns** [reply]
commit hash for old version :
f17444abcd647a299f23fe2bf6324b8947cdee22

**Reply by glenn-jocher** [reply]
Got it. Running this code to try to reproduce:

```bash
!git clone https://github.com/ultralytics/yolov5 yolov5-before
%cd yolov5-before
!git checkout f17444abcd647a299f23fe2bf6324b8947cdee22
!python train.py --data VOC.yaml --epochs 50 --cache --project discussion_9440 --name commit_f17444

%cd ..

!git clone https://github.com/ultralytics/yolov5 yolov5-master
%cd yolov5-master
!python train.py --data VOC.yaml --epochs 50 --cache --project discussion_9440 --master
```

**Reply by jiwoonjeon-mcns** [reply]
![result](https://user-images.githubusercontent.com/74587346/191912697-82d679e4-f938-4126-a510-5201abdf62eb.png)
I also come out the same way.
But, When I run it from where I installed it before, it comes out like this. Master's performance is the same as yours.
It seems to be my problem, so I'll try to find it. Thank you.

**Reply by glenn-jocher** [reply]
Great, thank you!

**Reply by jiwoonjeon-mcns** [reply]
in val.py
iouv = torch.linspace(0.1, 0.95, 10, device=device) <- before
iouv = torch.linspace(0.5, 0.95, 10, device=device) <- now
I found something different. After changing it, the performance is the same.
Was the previous one wrong?

**Comment by glenn-jocher** [comment]
@jiwoonjeon-mcns I trained with both commits and observe no change in performance. I'm unable to reproduce any problem with YOLOv5. If you have a **reproducible** issue please provide exact code to reproduce.

<img width="1257" alt="Screenshot 2022-09-22 at 21 40 33" src="https://user-images.githubusercontent.com/26833433/191836847-25943a1e-360c-4d01-8f21-3980c70cb399.png">

**Reply by glenn-jocher** [reply]
## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃
